### PR TITLE
[DPE-4985] Create fixture for HA cluster set up + break HA tests into separate groups

### PR DESCRIPTION
**Pull request by shayancanonical**
## Issue
https://mysqlcommunity.slack.com/archives/C9F7MGV1C/p1722604861637739
Seeing that when the freeze db test is run right after the kill db test, the cluster enters a really funky state where the `SIGSTOP`ed primary does not get booted from the cluster (even though it is `UNREACHABLE`) and a new primary is not elected

## Solution
We had plans to break down HA tests to run in separate runners. Create a fixture to set up a HA cluster, and break all HA tests into separate groups

### CODE REVIEW

File: charmcraft.yaml [hunk 1]
```diff
@@ -10,6 +10,10 @@ bases:
     channel: "22.04"
     architectures: [arm64]
 parts:
+  files:
+    plugin: nil
+    prime:
+      - scripts/log_rotate_dispatcher.py
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply charmcraft.yaml:16]: with dpw v18 (and pinning of charmcraft), i think this change is no longer needed

[the remark this answers is not in the document]
Comment (taurus-forever) [reply charmcraft.yaml:16]: We will need them to migrate to charmcraft 3. Prime will not be restored. dpw18 just ping charmcraft 2.x for now (until ARM release issue fixed).

**Review by taurus-forever — COMMENTED** [review]
IMHO, it is a good way forward! Thank you!
We still need to find out why group4 [failed on the first try](https://github.com/canonical/mysql-k8s-operator/actions/runs/10254250331/job/28369375218#step:28:572) of this PR... but step-by-step.

**Review by taurus-forever — APPROVED** [review]

**Review by paulomach — APPROVED** [review]
lovely

**Pull request merged**